game/lib/widget/progress_bar.dart

```diff
@@ -12,64 +12,111 @@ class LoadingBar extends StatelessWidget {
+ * 
+ */
+  String formatNumber(int number) {
+    if (number < 1000) {
+      return number.toString();
+    } else {
+      if (number == 1000) {
+        return "1K";
+      } else if (number > 1000 && number < 10000) {
+        return (number / 1000).toStringAsFixed(1) + 'K';
+      } else if (number == 10000) {
+        return "10K";
+      } else if (number > 10000 && number < 100000) {
+        return (number / 1000).toStringAsFixed(0) + 'K';
+      } else {
+        // Throw an exception for numbers greater than 100,000
+        throw FormatException('Number too large to format');
+      }
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Row(
-      children: [
-        Container(
-            width: MediaQuery.sizeOf(context).width * 0.5,
-            child: LayoutBuilder(
-                builder: (BuildContext context, BoxConstraints constraints) {
-              return Stack(children: [
-                Container(
-                  width: constraints.maxWidth,
-                  height: 13,
-                  alignment: Alignment.centerLeft,
-                  child: Container(
-                    decoration: new BoxDecoration(
-                      color: Color.fromARGB(255, 241, 241, 241),
-                      shape: BoxShape.rectangle,
-                      borderRadius: BorderRadius.all(Radius.circular(16.0)),
+    return Container(
+      constraints:
+          BoxConstraints(maxWidth: MediaQuery.sizeOf(context).width * 0.53),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        children: [
+          Expanded(
+              flex: 3,
+              child: LayoutBuilder(
+                  builder: (BuildContext context, BoxConstraints constraints) {
+                return Stack(children: [
+                  Container(
+                    width: constraints.maxWidth,
+                    height: 13,
+                    alignment: Alignment.centerLeft,
+                    child: Container(
+                      decoration: new BoxDecoration(
+                        color: Color.fromARGB(255, 241, 241, 241),
+                        shape: BoxShape.rectangle,
+                        borderRadius: BorderRadius.all(Radius.circular(16.0)),
+                      ),
                     ),
                   ),
-                ),
-                Container(
-                  width: (totalTasks > 0 ? tasksFinished / totalTasks : 0) *
-                      constraints.maxWidth,
-                  height: 13,
-                  alignment: Alignment.centerLeft,
-                  child: Container(
+                  Container(
+                    width: (totalTasks > 0 ? tasksFinished / totalTasks : 0) *
+                        constraints.maxWidth,
+                    height: 13,
+                    alignment: Alignment.centerLeft,
+                    child: Container(
```

Comment:
Instead of having a container with a another container inside of it, the BoxDeocration should be able to be applied to the outer container.